src/name.rs

```diff
@@ -234,6 +234,18 @@ impl Prefix {
             None
         }
     }
+
+    /// Return true if `self` is a sibling of `other` or any of `other`'s ancestors.
+    ///
+    /// E.g. siblings of ancestors of 000 are 001, 01 and 1.
+    pub fn is_sibling_of_ancestor_of(&self, other: &Prefix) -> bool {
+        if self.bit_count > other.bit_count {
+            return false;
+        }
+        let ancestor = Prefix::new(self.bit_count,
+                                   other.name.with_flipped_bit(self.bit_count - 1));
+        &ancestor == self
```

Comment:
Equivalently you could check that the common prefix length is `self.bit_count - 1`. Then you wouldn't need to create a new object.